extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -14,43 +14,415 @@
+        `${TAG} should contain exactly one <input> child`);
+    this.inputElement_ = inputElements[0];
+
+    this.filter_ = userAssert(this.element.getAttribute('filter'),
+        `${TAG} requires "filter" attribute.`);
+    userAssert(isEnumValue(FilterType, this.filter_),
+        `Unexpected filter: ${this.filter_}`);
+
+    this.minChars_ = this.element.hasAttribute('min-characters') ?
+      parseInt(this.element.getAttribute('min-characters'), 10) : 1;
+    this.maxEntries_ = parseInt(this.element.getAttribute('max-entries'), 10);
+
+    return this.mutateElement(() => {
+      this.container_ = this.createContainer_();
+      this.element.appendChild(this.container_);
+    });
+  }
+
+  /**
+   * Reads the 'items' data from the child <script> element.
+   * For use with static local data.
+   * @return {?Array}
+   * @private
+   */
+  getInlineData_() {
+    const scriptElements = childElementsByTag(this.element, 'SCRIPT');
+    if (!scriptElements.length) {
+      return null;
+    }
+    userAssert(scriptElements.length === 1,
+        `${TAG} should contain at most one <script> child`);
+    const scriptElement = scriptElements[0];
+    userAssert(isJsonScriptTag(scriptElement),
+        `${TAG} should be inside a <script> tag with type="application/json"`);
+    const json = tryParseJson(scriptElement.textContent,
+        error => {
+          throw error;
+        });
+    return json['items'] ? json['items'] : [];
+  }
+
+  /**
+   * Creates and returns <div> that contains the template-rendered children.
+   * @return {Element}
+   * @private
+   */
+  createContainer_() {
+    const container = this.element.ownerDocument.createElement('div');
+    container.classList.add('i-amphtml-autocomplete-results');
+    container.setAttribute('role', 'list');
+    toggle(container, false);
+    this.applyFillContent(container, /* replacedContent */ true);
+    return container;
   }
 
   /** @override */
   layoutCallback() {
-    // Actually load your resource or render more expensive resources.
-    return Promise.resolve();
+    // Disable autofill in browsers.
+    this.inputElement_.setAttribute('autocomplete', 'off');
+
+    // No static local data to filter against.
+    if (!this.inlineData_) {
+      return Promise.resolve();
+    }
+
+    // Register event handlers.
+    this.inputElement_.addEventListener('input', e => {
+      this.inputHandler_(e);
+    });
+    this.inputElement_.addEventListener('keydown', e => {
+      this.keyDownHandler_(e);
+    });
+    this.inputElement_.addEventListener('focus', () => {
+      this.showResults();
+    });
+    this.inputElement_.addEventListener('blur', () => {
+      this.hideResults();
+    });
+    this.container_.addEventListener('mousedown', e => {
+      this.selectHandler_(e);
+    });
+
+    return this.mutateElement(() => {
+      this.element.classList.add('i-amphtml-autocomplete');
+      this.inputElement_.classList.add('i-amphtml-autocomplete-input');
```

Comment:
do this in `buildCallback`

Reply:
I decided to remove these classes because their styling is covered by adding the blocks in `amp.css`